pkg/rbd/nodeserver.go

```diff
@@ -238,10 +244,14 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	targetPath := req.GetTargetPath()
 	isBlock := req.GetVolumeCapability().GetBlock() != nil
 	stagingPath := req.GetStagingTargetPath()
-	stagingPath += "/" + req.GetVolumeId()
+	volID := req.GetVolumeId()
+	stagingPath += "/" + volID
 
-	idLk := targetPathLocker.Lock(targetPath)
-	defer targetPathLocker.Unlock(idLk, targetPath)
+	if acquired := ns.VolumeLocks.TryAcquire(targetPath); !acquired {
```

Comment:
Same comment as the Publish/Unpublish locks in CephFS, change this to lock on the `VolID` instead, and safeguard against parallel Staging and Publish requests for the same volume.